mantidproject/mantid
# Replace QTableView with QTreeView

Description of work.

**To test:**
- Check code changes: relevant classes are `QDataProcessorTreeModel` (comes with unit tests), which replaces `QDataProcessorTableModel`. In `GenericDataProcessorPresenter`, check the actions (`insertRows()`, `insertGroups()`, `deleteRow()`, etc)
- Ensure that old TBL files can be loaded correctly (a sample TBL file and some sample datasets can be found here: [SampleData-ISIS.zip](https://github.com/mantidproject/mantid/files/361036/SampleData-ISIS.zip))
- Ensure that you can save a TBL file and load it again.
- Ensure runs can be reduced. For the reduction to work you will have to set `Params="-0.04"` in the textbox `Stitch1DMany`. Selecting runs (child items) will just reduce the selected runs. Selecting groups will reduce runs and post-process them.
- Check that actions work as you would expect. Actions can be accessed from the toolbar and from the outer menus. 
- Check changes to the documentation.

Fixes #16734 

Release notes were updated.

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
